opc-ua-sdk/sdk-client/src/main/java/org/eclipse/milo/opcua/sdk/client/subscriptions/OpcUaSubscriptionManager.java

```diff
@@ -351,6 +351,71 @@ private UInteger getLifetimeCount(double publishingInterval, UInteger maxKeepAli
         });
     }
 
+    @Override
+    public CompletableFuture<UaSubscription> transferSubscription(
+        UaSubscription subscription,
+        ItemTransferredCallback callback
+    ) {
+
+        OpcUaSubscription newSubscription = new OpcUaSubscription(
+            client,
+            subscription.getSubscriptionId(),
+            subscription.getRevisedPublishingInterval(),
+            subscription.getRevisedLifetimeCount(),
+            subscription.getRevisedMaxKeepAliveCount(),
+            subscription.getMaxNotificationsPerPublish(),
+            subscription.isPublishingEnabled(),
+            subscription.getPriority()
+        );
+
+        newSubscription.setRequestedLifetimeCount(subscription.getRequestedLifetimeCount());
+        newSubscription.setRequestedMaxKeepAliveCount(subscription.getRequestedMaxKeepAliveCount());
+        newSubscription.setRequestedPublishingInterval(subscription.getRequestedPublishingInterval());
+
+        subscriptions.put(newSubscription.getSubscriptionId(), newSubscription);
+
+        return newSubscription
+            .transferMonitoredItems(subscription, callback)
+            .thenApply(v -> newSubscription);
+    }
+
+    @Override
+    public CompletableFuture<UaSubscription> transferSubscription(
```

Comment:
I think this needs a `nextSequenceNumber` parameter, obtained from a `SubscriptionDiganosticsDataType` for the corresponding subscription. Without it the subscription manager will try to call republish for potentially many thousands of sequence numbers after receiving the first `PublishResponse` for the transferred subscription.

Reply:
I think the `transferSubscriptions` method returns `availableSequenceNumbers`. Since this is always called no matter the approach I think this would be better (assuming I'm on the right track here). Would the first first item in that array be sufficient? 

We would set the `lastSequenceNumber` value of the subscription with that number minus 1?